### PR TITLE
testing/wireguard: update to 0.0.20170320

### PR DESCRIPTION
**Pull request by itoffshore**
* wireguard-grsec
* wireguard-vanilla
* wireguard-tools

### CODE REVIEW

**Comment by jirutka** [comment]
Why is wiredguard split into three separated abuilds instead of one with three subpackages?

**Comment by itoffshore** [comment]
because 2 are kernel modules - it's packaged the same way as `zfs`

File: testing/wireguard-grsec/APKBUILD [hunk 1]
```diff
@@ -7,7 +7,7 @@ _kver=4.9.16
 _kpkgrel=0
 
 # when changing _ver we *must* bump _mypkgrel
-_ver=0.0.20170223
+_ver=0.0.20170320
```
Comment (ncopa) [inline testing/wireguard-grsec/APKBUILD]: comment says that when `_ver` changes we must increase _mypkgrel.

File: testing/wireguard-vanilla/APKBUILD [hunk 2]
```diff
@@ -7,7 +7,7 @@ _kver=4.9.16
 _kpkgrel=0
 
 # when changing _ver we *must* bump _mypkgrel
-_ver=0.0.20170223
+_ver=0.0.20170320
 _mypkgrel=0
```
Comment (ncopa) [inline testing/wireguard-vanilla/APKBUILD]: needs to be `_mypkgrel=1`

**Comment by ncopa** [comment]
3 packages is fine. The kernel modules should be built with kernel and the userspace tool should be separate so it can be shared with the different kernels.

I wonder if we should rename `wireguard-tools` to `wireguard`. Upstream source package is called `Wireguard`. We'd also need change the `install_if="wireguard-tools linux-grsec-$_kver"` to `
install_if="wireguard linux-grsec-$_kpkgver"` (the `$_kver` there should be `$_kpkgver` btw)

That way `apk add wireguard` would install the userspace package and the kernel package(s) for the installed kernel(s).

File: testing/wireguard-grsec/APKBUILD (hunk 1 above)
Comment (itoffshore) [inline testing/wireguard-grsec/APKBUILD]: fixed & rebased

File: testing/wireguard-vanilla/APKBUILD (hunk 2 above)
Comment (itoffshore) [inline testing/wireguard-vanilla/APKBUILD]: fixed & rebased

**Comment by itoffshore** [comment]
[Debian](https://packages.debian.org/search?searchon=names&keywords=wireguard) & [Arch](https://www.archlinux.org/packages/?sort=&q=wireguard&maintainer=&flagged=) have both named the userspace as `wireguard-tools` too.

**Comment by ncopa** [comment]
@itoffshore this failed on aarch64. Do you think you could report it upstream?

````
  CC [M]  /home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/compat/siphash/siphash.o
/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.c: In function 'curve25519':
/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.c:478:6: error: 'curve25519_use_avx' undeclared (first use in this function)
  if (curve25519_use_avx && irq_fpu_usable()) {
      ^~~~~~~~~~~~~~~~~~
/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.c:478:6: note: each undeclared identifier is reported only once for each function it appears in
/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.c:478:28: error: implicit declaration of function 'irq_fpu_usable' [-Werror=implicit-function-declaration]
  if (curve25519_use_avx && irq_fpu_usable()) {
                            ^~~~~~~~~~~~~~
/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.c:479:3: error: implicit declaration of function 'kernel_fpu_begin' [-Werror=implicit-function-declaration]
   kernel_fpu_begin();
   ^~~~~~~~~~~~~~~~
/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.c:480:3: error: implicit declaration of function 'curve25519_sandy2x' [-Werror=implicit-function-declaration]
   curve25519_sandy2x(mypublic, secret, basepoint);
   ^~~~~~~~~~~~~~~~~~
/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.c:481:3: error: implicit declaration of function 'kernel_fpu_end' [-Werror=implicit-function-declaration]
   kernel_fpu_end();
   ^~~~~~~~~~~~~~
/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.c: In function 'curve25519_generate_public':
/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.c:505:6: error: 'curve25519_use_avx' undeclared (first use in this function)
  if (curve25519_use_avx && irq_fpu_usable()) {
      ^~~~~~~~~~~~~~~~~~
/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.c:507:3: error: implicit declaration of function 'curve25519_sandy2x_base' [-Werror=implicit-function-declaration]
   curve25519_sandy2x_base(pub, secret);
   ^~~~~~~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:294: /home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src/crypto/curve25519.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:1490: _module_/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-4.9.17'
make: *** [Makefile:28: module] Error 2
make: Leaving directory '/home/buildozer/aports/testing/wireguard-vanilla/src/WireGuard-0.0.20170320.1/src'
>>> ERROR: wireguard-vanilla: all failed
````

**Comment by itoffshore** [comment]
ok will contact the mailing list & ping to @zx2c4

**Comment by zx2c4** [comment]
Known issue. It'll be fixed in the next snapshot. Sorry about that.

**Comment by zx2c4** [comment]
20170324 has been released, which fixes this issue.

**Comment by itoffshore** [comment]
@zx2c4 many thx for the prompt response - @ncopa  [PR to update to 0.0.20170324](https://github.com/alpinelinux/aports/pull/1082)

**Pull request closed without merging**